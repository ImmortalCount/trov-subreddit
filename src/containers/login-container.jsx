import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import LoginForm from '../components/login-form'
import handleLogin, { setUserData } from '../services/login'
import { authUser } from '../actions/user'

export class LoginContainer extends Component {
    constructor(props) {
        super(props)
        this.state = {
            isLoading: false,
            isError: false
        }
    }

    handleLoginSubmit = (creds) => {
        this.setState({
            isLoading: true,
            isError: false
        })
        handleLogin(creds, this.handleLoginError)
            .then((res) => {
                res.data ?
                    setUserData(res.data.token)
                    :
                    this.handleLoginError()
            })
            .catch(() => this.handleLoginError())
    }

    handleLoginError = () => {
        this.setState({
            isError: true,
            isLoading: false
        })
    }

    clearError = () => {
        this.setState({ isError: false })
    }

    render() {
        return (
            <LoginForm
                handleLoginSubmit={this.handleLoginSubmit}
                clearError={this.clearError}
                isError={this.state.isError}
                isLoading={this.state.isLoading}
            />
        )
    }
}

const mapStateToProps = state => ({
    authUser: state.authUser
})

const mapDispatchToProps = dispatch => (
    bindActionCreators({ authUser }, dispatch)
)

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(LoginContainer)
