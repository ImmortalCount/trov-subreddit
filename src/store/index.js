import { createStore, applyMiddleware, compose } from 'redux'
import promiseMiddleware from 'redux-promise'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import rootReducer from '../reducers'

const persistConfig = {
    key: 'root',
    storage: storage
}

const persistedReducer = persistReducer(persistConfig, rootReducer)

const enhancer = compose(
    applyMiddleware(
        promiseMiddleware
    ),
    window.window.__REDUX_DEVTOOLS_EXTENSION__ ?
        window.window.__REDUX_DEVTOOLS_EXTENSION__() :
        f => f
)

const store = createStore(persistedReducer, {}, enhancer)

export const persistor = persistStore(store)
export default store
