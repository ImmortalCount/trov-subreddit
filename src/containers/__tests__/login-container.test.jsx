import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import { LoginContainer } from '../login-container'
import store from '../../store'

describe('LoginContainer Component', () => {
    const defaultProps = {
        store: store,
        authUser: jest.fn()
    }

    const wrapper = shallow(
        <LoginContainer {...defaultProps} />
    )

    it('should render change stage when form is submitted', () => {
        const creds = {
            username: 'Emilio Estevez',
            password: 'foo'
        }
        wrapper.instance().handleLoginSubmit(creds)
        expect(wrapper.instance().state.isLoading).toBe(true)
        expect(wrapper.instance().state.isError).toBe(false)
    })

    it('should change state when handleLoginError is called', () => {
        wrapper.instance().handleLoginError()
        expect(wrapper.instance().state.isLoading).toBe(false)
        expect(wrapper.instance().state.isError).toBe(true)
    })

    it('should change state when clearError is called', () => {
        wrapper.instance().clearError()
        expect(wrapper.instance().state.isLoading).toBe(false)
        expect(wrapper.instance().state.isError).toBe(false)
    })

    it('LoginContainer matches snapshot', () => {
        const modalRender = renderer
            .create(<LoginContainer {...defaultProps} />)
            .toJSON()
        expect(modalRender).toMatchSnapshot()
    })
})
