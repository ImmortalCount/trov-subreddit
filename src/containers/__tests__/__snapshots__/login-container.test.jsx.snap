// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer Component LoginContainer matches snapshot 1`] = `
<div
  className="ui message login-form"
>
  <form
    disabled={false}
    noValidate={true}
    onError={[Function]}
    onFocus={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="ui form"
      onFocus={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label
          htmlFor="username"
        >
           
          Email Address
           
        </label>
        <div
          className="ui massive input"
        >
          <input
            id="username"
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="field"
      >
        <label
          htmlFor="password"
        >
           
          Password
           
        </label>
        <div
          className="ui massive input"
        >
          <input
            id="password"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            required={false}
            type="password"
            value=""
          />
        </div>
      </div>
      <button
        className="ui green button"
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
